src/timing.rs

```diff
@@ -93,6 +94,20 @@ impl Timeout {
     }
 }
 
+/// With exponential backoff, repeatedly try the callback until the result is `Ok`
+pub fn retry_until_ok<T, E, F: FnMut() -> Result<T, E>>(base: u64, mut callback: F) -> T {
+    let mut delay = base;
+    loop {
+        match callback() {
+            Ok(res) => return res,
+            Err(_) => {
+                sleep(Duration::from_millis(delay));
+                delay *= base;
```

Comment:
This is a very quick increase. I would instead use multiply by 2 (what we use for authorization attempts in the validator) and adding a maximum time to the config.